tests/test_restore_command.rs

```diff
@@ -100,3 +102,124 @@ fn test_restore() {
     R file1
     "###);
 }
+
+// Much of this test is copied from test_resolve_command
+#[test]
+fn test_restore_conflicted_merge() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(&test_env, &repo_path, "base", &[], &[("file", "base\n")]);
+    create_commit(&test_env, &repo_path, "a", &["base"], &[("file", "a\n")]);
+    create_commit(&test_env, &repo_path, "b", &["base"], &[("file", "b\n")]);
+    create_commit(&test_env, &repo_path, "conflict", &["a", "b"], &[]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @    conflict
+    ├─╮
+    o │  b
+    │ o  a
+    ├─╯
+    o  base
+    o
+    "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file")).unwrap()
+        , @r###"
+            <<<<<<<
+            %%%%%%%
+            -base
+            +a
+            +++++++
+            b
+            >>>>>>>
+            "###);
+    std::fs::write(repo_path.join("file"), "resolution").unwrap();
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["diff"]), 
+    @r###"
+    Resolved conflict in file:
+       1     : <<<<<<<
+       2     : %%%%%%%
+       3     : -base
+       4     : +a
+       5     : +++++++
+       6     : b
+       7     : >>>>>>>
+            1: resolution
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["restore", "file"]);
```

Comment:
nit: Add at least a blank line here to mark that this is the end of setup. A comment saying something like "// Restore the the conflicted state" might also be useful

Reply:
This is helpful, thanks! I have written some huge tests recently, and I'm not always sure how to make them readable.